// ng-qrcode/esm2015/lib/qr-code.module.js
import { CommonModule } from "@angular/common";
import { NgModule } from "@angular/core";
import { QrCodeComponent } from "./qr-code.component";
import { QrCodeDirective } from "./qr-code.directive";
import * as i0 from "@angular/core";
export class QrCodeModule {
}
QrCodeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: QrCodeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
QrCodeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: QrCodeModule, declarations: [QrCodeComponent,
        QrCodeDirective], imports: [CommonModule], exports: [QrCodeComponent,
        QrCodeDirective] });
QrCodeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: QrCodeModule, imports: [[
            CommonModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: QrCodeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        QrCodeComponent,
                        QrCodeDirective,
                    ],
                    imports: [
                        CommonModule,
                    ],
                    exports: [
                        QrCodeComponent,
                        QrCodeDirective,
                    ],
                }]
        }] });
